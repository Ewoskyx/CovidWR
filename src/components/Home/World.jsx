import React from 'react';
import PropTypes from 'prop-types';
import './Continent.css';
import { BsFillPeopleFill, BsClock } from 'react-icons/bs';
import { GiWorld, GiDeathSkull } from 'react-icons/gi';

const World = ({
  continent, title, countryCount, totalCountries, totalDeaths, updateDate,
}) => (
  <div className="col-12 col-md-10 continent-main">
    <div className="card continent-card text-white">
      <img className="card-img" src={continent} alt="Card" />
      <div className="card-img-overlay">
        <h5 className="card-title">{title}</h5>
        <p className="card-text">
          <GiWorld />
          {`  Available Countries: ${totalCountries}`}
        </p>
        <p className="card-text">
          <BsFillPeopleFill />
          {`  Confirmed Cases: ${countryCount}`}
        </p>
        <p className="card-text">
          <GiDeathSkull />
          {`  Total Deaths: ${totalDeaths}`}
        </p>
        <p className="card-text">
          <BsClock />
          {`  Last Update ${updateDate}`}
        </p>
      </div>
    </div>
  </div>
);

export default World;

World.propTypes = {
  continent: PropTypes.string,
  title: PropTypes.string,
  countryCount: PropTypes.string,
  totalCountries: PropTypes.number,
  totalDeaths: PropTypes.string,
  updateDate: PropTypes.string,
};

World.defaultProps = {
  continent: '',
  title: '',
  countryCount: '',
  totalCountries: 0,
  totalDeaths: '',
  updateDate: '',
};
