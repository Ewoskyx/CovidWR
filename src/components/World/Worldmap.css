.world-wrapper {
  border: none;
}

.worldmap {
  stroke: #999;
  stroke-width: 2px;
  stroke-linecap: round;
  stroke-linejoin: bevel;
}

.worldmap path {
  fill: #ccc;
  transition: fill 3.5s, ease;
}

.worldmap path:hover {
  filter: url(#myFilter);
  cursor: pointer;
  stroke-dasharray: 100;
  fill: #ececec;
  animation: pathAnimation 6s linear infinite;
}

@keyframes pathAnimation {
  to {
    stroke-dashoffset: 300;
  }
}
