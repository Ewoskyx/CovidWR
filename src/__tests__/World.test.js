import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import World from '../components/Home/World';
import store from '../redux/configStore';

const renderer = require('react-test-renderer');

it('Matches the snapshot of World component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <World />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders World component', () => {
  render(<Provider store={store}><BrowserRouter><World /></BrowserRouter></Provider>);
  const element = screen.getByText(/Available Countries/i);
  expect(element).toBeInTheDocument();
});
