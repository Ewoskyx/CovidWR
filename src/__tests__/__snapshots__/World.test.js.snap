// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot of World component 1`] = `
<div
  className="col-12 col-md-10 continent-main"
>
  <div
    className="card continent-card text-white"
  >
    <img
      alt="Card"
      className="card-img"
      src=""
    />
    <div
      className="card-img-overlay"
    >
      <h5
        className="card-title"
      >
        
      </h5>
      <p
        className="card-text"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M253.47 17.406C123.76 17.406 18.437 122.76 18.437 252.47c0 129.707 105.324 235.06 235.03 235.06 129.707 0 235.063-105.353 235.063-235.06 0-129.71-105.355-235.064-235.06-235.064zM367.874 68.75c61.246 38.19 101.97 106.14 101.97 183.72 0 17.143-1.993 33.823-5.75 49.81l-34.25-18.06 22 54.874c-9.454 21.647-22.362 41.432-38 58.687l-43.158-30.936-64.625 47.72-61.656 6.967-13.906-41.78-49.72 26.844-68.093-18.938 9.157 36.594c-28.41-21.793-51.23-50.466-66-83.563L81.25 304.47l32.25 17.124 59.22-9.875 2.843-40.908-37.344-1.718 4.905-17.844 30.78-25.313-25.093-15.625 67.22-38.593-45.345-29.657-66.625 40.187-49.437-15.28c13.812-32.14 35.21-60.22 61.906-82.064l-3.75 44.375 43.376-34.124 72 22.22-22.5-27.407L233 75.562l26.813 28.468 71 9.845-3.5-34.47 41.468 12.657-.905-23.312zm1.156 120.03L278 199.47l28.906 43.218 3.156 64.468L339.25 321l11.438-28.375 62.656 48.656L395.78 294l6.408-48.344-43.75-22.72 10.593-34.155zM221 192.438l-31.594 21.188 36.47 14.78 16.686-14.78L221 192.437zm22.188 144.688l18.687 52.594 19.78-42.564-38.467-10.03z"
          />
        </svg>
          Available Countries: 0
      </p>
      <p
        className="card-text"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
          />
          <path
            d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
            fillRule="evenodd"
          />
          <path
            d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
          />
        </svg>
          Confirmed Cases: 
      </p>
      <p
        className="card-text"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M255.997 16.004c-120 0-239.997 60-239.997 149.998C16 226.002 61 256 61 316c0 45-15 45-15 75 0 14.998 48.01 32.002 89.998 44.998v60h239.997v-60s90.567-27.957 90-45c-.933-27.947-15-30-15-74.998 0-30 45.642-91.42 44.998-149.998 0-90-119.998-149.998-239.996-149.998zm-90 179.997c33.137 0 60 26.864 60 60 0 33.136-26.863 60-60 60C132.863 316 106 289.136 106 256c0-33.136 26.862-60 59.998-60zm179.998 0c33.136 0 60 26.864 60 60 0 33.136-26.864 60-60 60-33.136 0-60-26.864-60-60 0-33.136 26.864-60 60-60zm-89.998 105c15 0 45 60 45 75 0 29.998 0 29.998-15 29.998h-60c-15 0-15 0-15-30 0-15 30-74.998 45-74.998z"
          />
        </svg>
          Total Deaths: 
      </p>
      <p
        className="card-text"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 3.5a.5.5 0 0 0-1 0V9a.5.5 0 0 0 .252.434l3.5 2a.5.5 0 0 0 .496-.868L8 8.71V3.5z"
          />
          <path
            d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm7-8A7 7 0 1 1 1 8a7 7 0 0 1 14 0z"
          />
        </svg>
          Last Update 
      </p>
    </div>
  </div>
</div>
`;
